samples/net/sockets/echo_server/boards/nucleo_g431rb.conf

```diff
@@ -0,0 +1,52 @@
+# Generic networking options
+CONFIG_NETWORKING=y
+CONFIG_NET_UDP=y
+CONFIG_NET_TCP=n
+CONFIG_NET_IPV6=n
+CONFIG_NET_IPV4=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETS_POSIX_NAMES=y
+CONFIG_POSIX_MAX_FDS=4
+CONFIG_NET_CONNECTION_MANAGER=y
+
+# Kernel options
+CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_ENTROPY_GENERATOR=y
+CONFIG_TEST_RANDOM_GENERATOR=y
+CONFIG_INIT_STACKS=y
+
+# Logging
+CONFIG_NET_LOG=n
+CONFIG_LOG=n
+CONFIG_NET_STATISTICS=n
+CONFIG_PRINTK=n
+
+# Network buffers
+CONFIG_NET_PKT_RX_COUNT=4
+CONFIG_NET_PKT_TX_COUNT=4
+CONFIG_NET_BUF_RX_COUNT=24
+CONFIG_NET_BUF_TX_COUNT=24
+CONFIG_NET_CONTEXT_NET_PKT_POOL=y
+
+# IP address options
+CONFIG_NET_MAX_CONTEXTS=2
+CONFIG_NET_MAX_CONN=2
+
+# Network shell
+CONFIG_NET_SHELL=n
+CONFIG_SHELL=n
+
+# Network application options and configuration
+CONFIG_NET_CONFIG_SETTINGS=y
+CONFIG_NET_CONFIG_NEED_IPV4=y
+CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.168.3.3"
+CONFIG_NET_CONFIG_PEER_IPV4_ADDR="192.168.3.1"
+
+# How many client can connect to echo-server simultaneously
+CONFIG_NET_SAMPLE_NUM_HANDLERS=1
+
+# LAN865x T1S driver specific
+CONFIG_NET_L2_ETHERNET=y
```

Comment:
This should be set in the board defaults and not here.

Reply:
This board don't have ethernet connector. This is required only when MicroE's add-in module is used.

Reply:
> This board don't have ethernet connector. This is required only when MicroE's add-in module is used.

Then should we have this patch at all as the other users of this board but without this add-in module would get incorrect configuration.

Reply:
There are "shields" defined for many other boards - like `MicroE ETH click` - `zephyr/boards/shields/mikroe_eth_click`

I could follow that path - the `nucleo_g431re` supports `arduino header`.

Although, in my case I only do have "wire" connected pins between `nucleo`  and `Micro BUS LAN8651` module, I can pretend that I do have a special arduino "shield" for this chip.

Reply:
> I can pretend that I do have a special arduino "shield" for this chip.

No, you can't. We only maintain actual shields.